pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/ReplicatedSubscriptionSnapshotCache.java

```diff
@@ -33,29 +35,96 @@
 public class ReplicatedSubscriptionSnapshotCache {
     private final String subscription;
     private final NavigableMap<Position, ReplicatedSubscriptionsSnapshot> snapshots;
+    // Used to record the timestamp of snapshots location, which will be used to adjust cache update frequency later.
+    private final NavigableMap<Position, Long> positionToTimestamp;
```

Comment:
instead of adding a separate data structure for timestamps, add a new Java record to hold `ReplicatedSubscriptionsSnapshot` and the timestamp. Modify the `snapshots` to use this type of value.

Reply:
After making this change, I'll review the code again since the current solution of having both `snapshots` and `positionToTimestamp` makes the code harder to understand. It will be easier to add more review comments after this is addressed.

Reply:
@liangyepianzhou You didn't seem to follow the advice about adding a Java record. Since ReplicatedSubscriptionsSnapshot is a protobuf message, you shouldn't be adding fields there since the `publishTime` is outside of it. Adding a field just to be able to solve holding the timestamp wouldn't be the correct approach.

Instead, add a Java `record` that is just private to this `ReplicatedSubscriptionSnapshotCache`:

```java
private record SnapshotEntry(ReplicatedSubscriptionsSnapshot snapshot, long timestamp) {
}
```